napari_allencell_annotator/_tests/widgets/template_list_test.py

```diff
@@ -1,13 +1,124 @@
 from unittest import mock
 from unittest.mock import MagicMock, patch, create_autospec
 
+from PyQt5.QtWidgets import QPushButton, QLineEdit
+
+from napari_allencell_annotator.model.combo_key import ComboKey
+
+from napari_allencell_annotator.model.key import Key
+from napari_allencell_annotator.widgets.template_item import ItemType
+
 from napari_allencell_annotator.widgets.template_list import (
     TemplateList,
     TemplateItem,
     QSpinBox,
     QCheckBox,
     QComboBox,
 )
+import pytest
+
+
+@pytest.fixture()
+def template_list(qtbot) -> TemplateList:
+    return TemplateList()
+
+
+def test_add_item_string(template_list: TemplateList) -> None:
+
+    # ACT
+    template_list.add_item("test_string", Key("string", "test_value"))
+
+    # ASSERT
+    assert template_list.count() == 1
+
+    item: TemplateItem = template_list.item(0)
+    assert item.name.text() == "test_string"
+    assert item.type == ItemType.STRING
+    assert item.default == "test_value"
+    assert isinstance(item.editable_widget, QLineEdit)
+    assert item.editable_widget.text() == "test_value"
+
+    # test template_list.items
+    assert len(template_list.items) == 1
+    assert template_list.items[0] == item
+
+
+def test_add_item_number(template_list: TemplateList) -> None:
+    # ACT
+    template_list.add_item("test_number", Key("number", 0))
+
+    # ASSERT
+    assert template_list.count() == 1
+
+    item: TemplateItem = template_list.item(0)
+    assert item.name.text() == "test_number"
+    assert item.type == ItemType.NUMBER
+    assert item.default == 0
+    assert isinstance(item.editable_widget, QSpinBox)
+    assert item.editable_widget.value() == 0
+
+    # test template_list.items
+    assert len(template_list.items) == 1
+    assert template_list.items[0] == item
+
+
+def test_add_item_bool(template_list: TemplateList) -> None:
+    # ACT
+    template_list.add_item("test_bool", Key("bool", True))
+
+    # ASSERT
+    assert template_list.count() == 1
+
+    item: TemplateItem = template_list.item(0)
+    assert item.name.text() == "test_bool"
+    assert item.type == ItemType.BOOL
+    assert item.default == True
+    assert isinstance(item.editable_widget, QCheckBox)
+    assert item.editable_widget.isChecked()
+
+    # test template_list.items
+    assert len(template_list.items) == 1
+    assert template_list.items[0] == item
+
+
+def test_add_item_list(template_list: TemplateList) -> None:
+    # ACT
+    template_list.add_item("test_list", ComboKey(list, ["a", "b"], "a"))
+
+    # ASSERT
+    assert template_list.count() == 1
+
+    item: TemplateItem = template_list.item(0)
+    assert item.name.text() == "test_list"
+    assert item.type == ItemType.LIST
+    assert item.default == "a"
+    assert isinstance(item.editable_widget, QComboBox)
+    assert item.editable_widget.itemText(0) == "a"
+    assert item.editable_widget.itemText(1) == "b"
+    assert item.editable_widget.currentText() == "a"
```

Comment:
not needed, you're testing beyond TemplateList

Reply:
removed in https://github.com/aics-int/napari-allencell-annotator/pull/92/commits/7f445d5488a221349dad061156ede36c7ed49a37